### PR TITLE
(#148) On This Page tests 

### LINKED ISSUE #148: Story: On this page section tests
## As a user, I want to be able to use On this Page links to easily navigate between page sections

> **ESTIMATE** TBD

### Acceptance Criteria

```gherkin
Scenario Outline: On this page section displays on article and PDQ CIS
  Given user is navigating to "<url>"
    Then "<otpTitle>" is displayed
 And OTP links have the following "<titles>" and "<hrefs>"
 
Examples:
|content type|url|titles|hrefs|otpTitle|
|article english|/about-cancer/treatment/side-effects/|What is...,How to...|#what-is,#how-to|On This Page|
|article spanish|/espanol/hoja-informativa-reconstruccion|¿Qué es la reconstrucción del seno?,¿Cómo usan los cirujanos implantes para reconstruir el seno de una mujer?,¿Cómo usan los cirujanos el tejido del cuerpo mismo de la mujer para reconstruir su seno?|#qu-es-la-reconstruccin-del-seno,#cmo-usan-los-cirujanos-implantes-para-reconstruir-el-seno-de-una-mujer,#cmo-usan-los-cirujanos-el-tejido-del-cuerpo-mismo-de-la-mujer-para-reconstruir-su-seno|En Esta Página|
|pdq Patient version EN| /types/lung/patient/lung-prevention-pdq|What is prevention?,General Information About Lung Cancer,Lung Cancer Prevention,About This PDQ Summary|#_29,#_4,#_12,#_AboutThis_1|On This Page|
|pdq HP version EN| /types/lung/hp/lung-prevention-pdq|Overview,Description of the Evidence,Changes to This Summary (03/02/2018),About This PDQ Summary|#_171,#_16,#_80,#_AboutThis_1|On This Page|
## TODO add patient spanish and hp spanish PDQ examples following the pattern above
```

```gherkin
Scenario Outline: On this page section displays as an accordion on mobile breakpoint
Given screen breakpoint is set to "mobile"
  When user is navigating to "<url>"
 Then on this page section title is not displayed
 And on this page accordion is displayed
And all of the accordion items are collapsed
 
Examples:
|content type|url|
|article english|/about-cancer/treatment/side-effects/|
|article spanish|/espanol/hoja-informativa-reconstruccion|
|pdq Patient version EN| /types/lung/patient/lung-prevention-pdq|
|pdq HP version EN| /types/lung/hp/lung-prevention-pdq|
## TODO add patient spanish and hp spanish PDQ examples 
```
```gherkin
Scenario Outline: OTP section does not exist on any content types, beside article and pdq
Given user is navigating to "<url>"
Then On This page section does not exist

Examples:
|url|
## TODO add one url for each content type other than article and pdq|
```
### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by alinai**
Added Feature File
Closes #148

### CODE REVIEW

**Comment by alinai** [comment]
Test Results:
image

@KateMashkinaNIH , please review. Thanks!!

File: cypress/integration/Crosscutting/OnThisPageTest/index.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/// <reference types="Cypress" />
+
+Then('{string} is displayed', (otpTitle) => {
+    cy.get('.on-this-page').contains(otpTitle).should('be.visible');
+});
+
+And('OTP links have the following {string}', (options) => {
+    const expectedOptions = options.split(',');
+    cy.log(expectedOptions);
```
Comment (kate-mashkina) [inline cypress/integration/Crosscutting/OnThisPageTest/index.js:9]: remove console.log()

File: cypress/integration/Crosscutting/OnThisPageTest/index.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    for (let i = 0; i < expectedOptions.length; i++) {
+        cy.get('#cgvBody section h2').eq(i).invoke('text').then((text) => {
+            expect(text.trim()).equal(expectedOptions[i]);
+        });
+    }
+});
+
+And('OTP links have the {string} as follows', (options) => {
+    const expectedOptions = options.split(',');
+    cy.log(expectedOptions);
```
Comment (kate-mashkina) [inline cypress/integration/Crosscutting/OnThisPageTest/index.js:19]: remove console.log()

**Review by kate-mashkina — CHANGES_REQUESTED** [review]

File: cypress/integration/Crosscutting/OnThisPageTest/index.js (hunk 1 above)
Comment (alinai) [reply cypress/integration/Crosscutting/OnThisPageTest/index.js:9]: Removed

File: cypress/integration/Crosscutting/OnThisPageTest/index.js (hunk 2 above)
Comment (alinai) [reply cypress/integration/Crosscutting/OnThisPageTest/index.js:19]: Removed

**Comment by alinai** [comment]
Also removed 3 Application Page urls. Test results after updated per review:

image

**Review by KateMashkinaNIH — APPROVED** [review]

**Pull request merged**